server/controllers/genAiController.ts

```diff
@@ -0,0 +1,59 @@
+import createAI from '@/server/utils/AIFactory';
+import { createError } from '@/server/middleware/errorHandling';
+
+import type { Request, Response, NextFunction } from 'express';
+import { type GenerativeAIModel } from '@server/types/index';
```

Comment:
Not that this really matters that much, but should the 'type' be outside of the  brackets?